### PR TITLE
Fixing `RPM` algorithm

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|#23818|

### Objective

This PR aims to resolve the discrepancies in RPM package version matching by removing the truncation of the release part in our comparison algorithm. This adjustment aligns our method more closely with the original RPM algorithm, ensuring higher accuracy in package management operations. The unit tests have also been updated to reflect these changes.

### Changes Implemented
- **Algorithm Adjustment**:
  - Removed the truncation process from the RPM comparison algorithm in `rpmVersionCompare.cpp`.
  
- **Unit Tests (UTs) Adjustment**:
  - Updated the unit tests to accommodate the new algorithm without the release truncation.
  - Added new test cases that specifically check for edge cases that were previously mishandled due to truncation.

### Implementation Details
- Revisited all unit tests related to RPM version comparison to ensure they accurately test the revised logic under various scenarios, including complex version strings and boundary conditions.

### Testing
- Implemented new unit tests to cover scenarios involving empty release strings, particularly where truncation would have previously altered outcomes.

-------------------------------------

### CODE REVIEW

**Review by sebasfalcone — APPROVED** [review]

**Review by sebasfalcone — APPROVED** [review]

File: src/wazuh_modules/vulnerability_scanner/tests/unit/versionMatcher_test.cpp [hunk 1]
```diff
@@ -700,8 +700,8 @@ TEST_F(VersionMatcherTest, compareRpmVer_rmpPackages)
                                       "1.0.1-10.module_el8.5.0+150+5f0dbea0.alma.ppc64le",
                                       VersionObjectType::RPM),
               VersionComparisonResult::A_EQUAL_B); // Alma
-    EXPECT_EQ(VersionMatcher::compare("2.2.20-2.el8.centos", "2.2.20-2.el8", VersionObjectType::RPM),
-              VersionComparisonResult::A_EQUAL_B); // CentOS
+    EXPECT_EQ(VersionMatcher::compare("2.2.20", "2.2.20-2.el8", VersionObjectType::RPM),
+              VersionComparisonResult::A_LESS_THAN_B); // No release
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/tests/unit/versionMatcher_test.cpp:704]: Lets keep this test with the expected behaviour

```suggestion
    EXPECT_EQ(VersionMatcher::compare("2.2.20", "2.2.20-2.el8", VersionObjectType::RPM),
              VersionComparisonResult::A_LESS_THAN_B); // No release
    EXPECT_EQ(VersionMatcher::compare("2.2.20-2.el8.centos", "2.2.20-2.el8", VersionObjectType::RPM),
              VersionComparisonResult::A_GREATER_THAN_B); // CentOS
```

Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/unit/versionMatcher_test.cpp:704]: `2.2.20-2.el8.centos` is a little probable version, the correct way should be `2.2.20-2.centos`. We should consider? I think is covered by another test

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/019/Readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+# Platforms
+
+## RedHat 9.3 (Pow)
+
+- Input events
+  - ![001](input_001.json)
+  - ![002](input_002.json)
+  - ![003](input_003.json)
+  - ![004](input_004.json)
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/019/Readme.md:19]:
```suggestion
  - [001](input_001.json)
  - [002](input_002.json)
  - [003](input_003.json)
  - [004](input_004.json)
```

File: src/wazuh_modules/vulnerability_scanner/qa/test_data/019/input_004.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "name": "python3-libs",
+    "priority": "optional",
+    "scan_time": "2023/08/04 19:56:11",
+    "size": 72,
+    "source": "python3-libs",
+    "vendor": "Red Hat, Inc.",
+    "version": "3.9.18-1.el9_3"
+  },
+  "operation": "INSERTED"
+}
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data/019/input_004.json:23]:
```suggestion
}

```

**Review by sebasfalcone — APPROVED** [review]
LGTM!
